src/devices/Sn74hc595/GpioControllerExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Linq;
+
+/// <summary>
+/// Extension methods for GpioController
+/// </summary>
+public static class GpioControllerExtensions
+{
+    /// <summary>
+    /// Opens a set of pins.
+    /// </summary>
+    public static void OpenPins(this GpioController controller, PinMode mode, params int[] pins)
+    {
+        foreach (var pin in pins)
+        {
+            controller.OpenPin(pin, mode);
+        }
+    }
+
+    /// <summary>
+    /// Opens a pin as PineMode.Output and writes value.
+    /// </summary>
+    public static void OpenPinAndWrite(this GpioController controller, int pin, PinValue value)
+    {
+        controller.OpenPin(pin, PinMode.Output);
+        controller.Write(pin, value);
+    }
+
+    /// <summary>
+    /// Writes a value to multiple pins.
+    /// </summary>
+    public static void WriteValueToPins(this GpioController controller, int value, params int[] pins)
+    {
+        for (int i = 0; i < pins.Length; i++)
+        {
+            controller.Write(pins[i], value);
+        }
+    }
+
+    /// <summary>
+    /// Writers multiple values to the same pin.
+    /// </summary>
+    public static void WriteValuesToPin(this GpioController controller, int pin, params int[] values)
```

Comment:
What is the purpose of this?

Reply:
This API is the opposite use case of the other extension you called out. The first one writes the same value to multiple pins. This one writes multiples values to a single pin. In this case, this how I manage the leading and falling edge of the pin (which is required by the protocol).

Use: https://github.com/richlander/iot/blob/Sn74hc595/src/devices/Sn74hc595/Sn74hc595.cs#L84

Reply:
I had the same question when I saw it at first but so with your answer I take that this is so that you get an event of some sorts by doing a 0 to 1 or 1 to 0?

Reply:
Yes,